src/redis_info_provider/info_servicer.py

```diff
@@ -58,11 +58,20 @@ def _filter_info(cls, full_info, keys):
         :param keys: A set of strings to match for.
         """
 
+        def is_matching_key(full_key, keys, exact=True):
+            if exact:
+                return full_key in keys
+            else:
+                for key in keys:
+                    if key in full_key:
```

Comment:
yes, but `startswith` is not more CPU-intensive than `==`

i will use `startswith`, but then maybe instead of the `exact` arg, which can be ambiguous, maybe I will use an arg called `use_prefix_matching`? WDYT?

Reply:
Well, not quite. The original code was
```python
{k: full_info[k] for k in keys if k in full_info}
```
which is O(k) in the length of `keys`, because membership checks are O(1) average-case.

Now we're comparing every key in `full_info` to every key in `keys`, so the new code will be O(k)*O(n) if _n_ is the number of INFO keys. Granted, the constant multiplier we had here for the regex implementation was probably considerably higher than a simple `startswith`, but it's still at least an order of magnitude change in number of operations.

On a different note, `use_prefix_matching` (or just `prefix_matching`) sounds good as a name.